src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -393,40 +397,44 @@ public String getStringByName(String memberName) {
     public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
             @Nonnull Launcher launcher, @Nonnull TaskListener listener)
             throws InterruptedException, IOException {        
-        final boolean isLinuxLauncher = launcher.isUnix();
+        //Set the environment variable specific to the this build
+        setEnv(build.getEnvironment(listener));
+        nodeSpecificfileSeparator = getNodeSpecificFileSeperator(launcher);
         
         // Invoke MATLAB command and transfer output to standard
         // Output Console
 
-        buildResult = execMatlabCommand(build, workspace, launcher, listener, isLinuxLauncher);
+        buildResult = execMatlabCommand(workspace, launcher, listener);
 
         if (buildResult != 0) {
             build.setResult(Result.FAILURE);
         }
     }
 
-    private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace, Launcher launcher,
-            TaskListener listener, boolean isLinuxLauncher)
+    private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
+            TaskListener listener)
             throws IOException, InterruptedException {
-        setEnv(build.getEnvironment(listener));
-        final String testRunMode = this.getTestRunTypeList().getDescriptor().getId();
-        
-        // Copy MATLAB scratch file into the workspace only if Automatic option is selected.
-        if (testRunMode.contains(AUTOMATIC_OPTION)) {
-            copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE,
-                    workspace, getClass().getClassLoader());
-        }
         ProcStarter matlabLauncher;
         try {
-            MatlabReleaseInfo rel = new MatlabReleaseInfo(getLocalMatlab());
+            FilePath nodeSpecificMatlabRoot = new FilePath(launcher.getChannel(),getLocalMatlab());
```

Comment:
similar comment as earlier. If the only way the local matlab is usable is by wrapping in a FilePath object, then just have it give a FilePath object.

Reply:
where mostly getLocalMatlab() is used wrapped in FilePath(). its not always true. getLocalMatlab()  is wrapped in FilePath only when we want to now the versions of the MATLAB. otherside in some cases we just would just require getLocalMAtlab() out put with different way of FilPath construction. for eg this case needs FilePath construction with Launcher  and file as string . everywhere else where we just need expanded Matlabroot, we will just use getLocalMatlab() as is.